netmon/netmon.go

```diff
@@ -0,0 +1,661 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log/syslog"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	lSyslog "github.com/sirupsen/logrus/hooks/syslog"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+// The following types and structures have to be kept in sync with the
+// description of the agent protocol. Those definitions need to be in their
+// own separate package so that they can be imported directly from this code.
+// The reason for not importing them now, is because importing the whole agent
+// protocol adds up too much overhead because of the grpc protocol involved.
+
+// IPFamily define the IP address family type.
+type IPFamily int32
+
+// IPAddress describes the IP address format expected by Kata API.
+type IPAddress struct {
+	Family  IPFamily `json:"family,omitempty"`
+	Address string   `json:"address,omitempty"`
+	Mask    string   `json:"mask,omitempty"`
+}
+
+// Interface describes the network interface format expected by Kata API.
+type Interface struct {
+	Device      string       `json:"device,omitempty"`
+	Name        string       `json:"name,omitempty"`
+	IPAddresses []*IPAddress `json:"IPAddresses,omitempty"`
```

Comment:
Capitalization? OK.

Reply:
Oh! You mean from the json definition?
Yes I'll change that!

Reply:
Yes. Thanks.

Reply:
Oh wait, I did that so that it comply with the structure defined in `agent/protocols/grpc/`. I'm afraid the unmarshalling might not work if I change this.